eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/SSSplitWriter.java

```diff
@@ -0,0 +1,295 @@
+import uk.ac.ebi.eva.accession.clustering.metric.ClusteringMetric;
+import uk.ac.ebi.eva.accession.core.model.ISubmittedVariant;
+import uk.ac.ebi.eva.accession.core.model.dbsnp.DbsnpSubmittedVariantEntity;
+import uk.ac.ebi.eva.accession.core.model.dbsnp.DbsnpSubmittedVariantOperationEntity;
+import uk.ac.ebi.eva.accession.core.model.eva.SubmittedVariantEntity;
+import uk.ac.ebi.eva.accession.core.model.eva.SubmittedVariantInactiveEntity;
+import uk.ac.ebi.eva.accession.core.model.eva.SubmittedVariantOperationEntity;
+import uk.ac.ebi.eva.accession.core.service.nonhuman.SubmittedVariantAccessioningService;
+import uk.ac.ebi.eva.metrics.metric.MetricCompute;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.springframework.data.mongodb.core.query.Criteria.where;
+import static org.springframework.data.mongodb.core.query.Query.query;
+
+public class SSSplitWriter implements ItemWriter<SubmittedVariantEntity> {
+
+    private static final Logger logger = LoggerFactory.getLogger(SSSplitWriter.class);
+
+    private final String assembly;
+
+    private final ClusteringWriter clusteringWriter;
+
+    private final SubmittedVariantAccessioningService submittedVariantAccessioningService;
+
+    private static final String ID_ATTRIBUTE = "_id";
+
+    private final MongoTemplate mongoTemplate;
+
+    private final List<SubmittedVariantOperationEntity> allSplitCandidatesForCurrentBatch = new ArrayList<>();
+
+    private final MetricCompute<ClusteringMetric> metricCompute;
+
+    public SSSplitWriter(String assembly, ClusteringWriter clusteringWriter,
+                         SubmittedVariantAccessioningService submittedVariantAccessioningService,
+                         MongoTemplate mongoTemplate, MetricCompute<ClusteringMetric> metricCompute) {
+        this.assembly = assembly;
+        this.clusteringWriter = clusteringWriter;
+        this.submittedVariantAccessioningService = submittedVariantAccessioningService;
+        this.mongoTemplate = mongoTemplate;
+        this.metricCompute = metricCompute;
+    }
+
+    @Override
+    public void write(@Nonnull List<? extends SubmittedVariantEntity> submittedVariantEntities)
+            throws MongoBulkWriteException, AccessionCouldNotBeGeneratedException, InstantiationException,
+            IllegalAccessException {
+        // We do this to avoid tedious generic type signatures in the method calls
+        List<SubmittedVariantEntity> svesWithDuplicateID = new ArrayList<>(submittedVariantEntities);
+
+        // All the following steps are designed to be idempotent
+        registerSplitCandidates(svesWithDuplicateID);
+        processSplitCandidates(this.allSplitCandidatesForCurrentBatch);
+        removeSplitCandidates(this.allSplitCandidatesForCurrentBatch);
```

Comment:
I'm not sure I understand why it is necessary to write candidates to the DB to remove them again right after the processing of the batch is complete.

Reply:
See [comment](https://github.com/EBIvariation/eva-accession/pull/342#discussion_r880505952)